src/web5.js

```diff
@@ -108,25 +109,29 @@ class Web5 extends EventTarget {
    * @param {*} request.data - The message data (if any).
    * @param {object} request.message - The DWeb message.
    * @param {string} request.target - The DID to send the message to.
-   * @returns Promise
+   * @returns {Promise<Web5SendResponse>}
    */
   async #send(endpoints, request) {
-    let response;
+    let response, message = {};
     for (let endpoint of endpoints) {
       try {
-        const url = parseURL(endpoint);
+        const url = parseUrl(endpoint);
         response = await this.#transports[url?.protocol?.slice(0, -1)]?.send(url.href, request);
       } catch (error) {
-        console.log(error);
+        console.error(error);
         // Intentionally ignore exception and try the next endpoint.
       }
       if (response) break; // Stop looping and return after the first endpoint successfully responds.
     }
 
-    return response ?? { status: { code: 503, detail: 'Service Unavailable' } };
+    if (!isUnsignedMessage(request.message)) {
+      // If the message is signed return the `descriptor`, and if present, `recordId`.
+      const { recordId = null, descriptor } = request.message.message;
+      message = { recordId, descriptor };
+    }
+
+    response ??= { status: { code: 503, detail: 'Service Unavailable' } };
+
+    return { message, ...response };
```

Comment:
do we want to allow for `message` to be overridden by a `response.message`?

Reply:
Yes, this was intentional.   The message property would either be present in `response` or created by the conditional just above the highlighted lines.  If it was present in the response, `message` would be an empty object, and therefore, should be overwritten.

However, declaring `message` as an empty object and only conditionally setting properties was not necessary.  I refactored the code in 78d637d1e0646c06fae6ec3b0fd7dbac930bd995